### PR TITLE
feat(generation): generate playgrounds for guides

### PR DESCRIPTION
**Pull request by mapsandapps**
See FW-4717 and [design doc update](https://github.com/ionic-team/ionic-framework-design-documents/pull/149) for extra context.

Previously, you couldn't create a playground for animations, gestures, etc., without using some workarounds. Now the prompts ask if you're creating a playground for a component or not and adjust the other prompt questions accordingly.

⚠️ I also did a little refactoring & renaming to make the code more clear. Walking through the commits one-by-one may help for review.

I included an `ion-card` in the templates, since the animation playgrounds tend to have those.

### CODE REVIEW

File: _templates/playground/new/index.js [hunk 1]
```diff
@@ -5,60 +5,80 @@ const changeCase = require('change-case');
...
+                    return value.match(/^ion-[a-z/-]*[a-z]+$/)
+                      ? true
+                      : "Component name must be kebab-case and begin with 'ion-'";
+                  },
+                }
+              : {
+                  type: 'input',
+                  name: 'name',
+                  message: 'What is this playground for?',
+                  initial: 'animations',
```
Comment (brandyscarney) [inline _templates/playground/new/index.js:37]: Should we add some validation here like the others to make sure only letters are entered?

File: _templates/playground/new/index.js [hunk 2]
```diff
@@ -5,60 +5,80 @@ const changeCase = require('change-case');
...
+                  validate(value) {
+                    return value.match(/^ion-[a-z/-]*[a-z]+$/)
+                      ? true
+                      : "Component name must be kebab-case and begin with 'ion-'";
+                  },
+                }
+              : {
+                  type: 'input',
+                  name: 'name',
+                  message: 'What is this playground for?',
```
Comment (brandyscarney) [inline _templates/playground/new/index.js:36]:
```suggestion
                  message: 'Which guide section is this playground for?',
```

I think it's probably safe to say this has to go in a guide if it isn't a component, right?

**Review by brandyscarney — CHANGES_REQUESTED** [review]

File: _templates/playground/new/index.js (hunk 1 above)
Comment (mapsandapps) [reply _templates/playground/new/index.js:37]: Good idea! Done in 4f4abb3.

**Review by brandyscarney — APPROVED** [review]
Looks great!

**Pull request merged**